lnwallet/channel.go

```diff
@@ -6738,17 +6741,6 @@ func (lc *LightningChannel) validateFeeRate(feePerKw chainfee.SatPerKWeight) err
 			newFee, baseBalance)
 	}
 
-	// If this new balance is below our reserve, then we can't accommodate
-	// the fee change, so we'll reject it.
-	balanceAfterFee := baseBalance - newFee
-	if balanceAfterFee.ToSatoshis() < lc.channelState.LocalChanCfg.ChanReserve {
-		return fmt.Errorf("cannot apply fee_update=%v sat/kw, "+
-			"new balance=%v would dip below channel reserve=%v",
-			int64(feePerKw),
-			balanceAfterFee.ToSatoshis(),
-			lc.channelState.LocalChanCfg.ChanReserve)
-	}
-
```

Comment:
I have removed this since i dont think it is possible for the `balanceAfterFee` to be less than our local reserve amount. The `availableBalance` already takes into account our reserve amount and so we would actually need to add the reserve amount to balanceAfterFee for this check to mean anything. And if we did so, then then only way this check would fail is if `baseBalance - newFee` was negative, in which case, the previous check would have errored.

Reply:
IIRC we added this to guard against an actual issue we encountered in the wild. Would prefer to leave it in place based on that pretext: https://github.com/lightningnetwork/lnd/commit/3e90000e2af2cd8212762e4de6419a43977d7984 

Also a quick log search on my mainnet node shows that hits is indeed being hit in the wild: 
```
$ ack 'dip below' .  | wc
 68    1564   14675
```

Reply:
This check was added before the change in [5e89d5b](https://github.com/lightningnetwork/lnd/commit/5e89d5b) which adds [this](https://github.com/lightningnetwork/lnd/blob/master/lnwallet/channel.go#L6656). So the availableBalance returned now already has the reserve amount subtracted from it which was not the case when this check was originally added. I think the reason why the log is currently still being hit is because `balanceAfterFee` already has the reserve amount subtracted from it so it is very possible that this balance is below reserve, but that is ok cause reserve was already subtracted. So i dont think this check makes sense unless `balanceAfterFee` is changed to `baseBalance - newFee + chanReserve`. which will always be >= chanReserve.